packages/firestore/test/unit/local/query_engine.test.ts

```diff
@@ -307,8 +387,9 @@ describe('QueryEngine', () => {
 
     // Add a query mapping for a document that matches, but that sorts below
     // another document due to a pending write.
-    await addDocument(PENDING_MATCHING_DOC_A);
-    await persistQueryMapping(PENDING_MATCHING_DOC_A.key);
+    await addDocument(MATCHING_DOC_A);
+    await addMutation(patchMutation('coll/a', { order: 1 }));
```

Comment:
These changes are to support IndexedDb persistence. We cannot store a document with mutations in indexeddb so we instead have to store a document and its mutation.